documentation/modules/cruise-control/con-optimization-goals.adoc

```diff
@@ -0,0 +1,133 @@
+. Leader replica distribution
+. Preferred leader election
+//. Intra-broker disk capacity
+//. Intra-broker disk usage distribution   
+
+For more information on each optimization goal, see link:https://github.com/linkedin/cruise-control/wiki/Pluggable-Components#goals[Goals^] in the Cruise Control Wiki.
+
+NOTE: CPU goals, intra-broker disk goals, "Write your own" goals, and Kafka assigner goals are not yet supported.
+
+As described in the following sections, you can customize the supported optimization goals by reordering them in terms of priority, and disabling goals to exclude from optimization proposal calculations.
+
+[discrete]
+== Goals configuration in {ProductName}
+
+You configure optimization goals in the `Kafka` and `KafkaRebalance` custom resources. Cruise Control has configurations for link:#master-goals[master], link:#default-goals[default], and link:#user-provided-goals[user-provided] optimization goals, as well as link:#hard-soft-goals[hard] optimization goals that must be satisfied.
+
+The following sections describe each goal configuration in more detail.
+
+[id="master-goals"]
+[discrete]
+=== Master optimization goals
+
+The _master optimization goals_ are available to all users.
+Unless you change the Cruise Control xref:proc-deploying-cruise-control-{context}[deployment configuration], {ProductName} will inherit the following master optimization goals from Cruise Control, in descending priority order:
+
+[source]
+RackAwareGoal; ReplicaCapacityGoal; DiskCapacityGoal; NetworkInboundCapacityGoal; NetworkOutboundCapacityGoal; CpuCapacityGoal; ReplicaDistributionGoal; PotentialNwOutGoal; DiskUsageDistributionGoal; NetworkInboundUsageDistributionGoal; NetworkOutboundUsageDistributionGoal; CpuUsageDistributionGoal; TopicReplicaDistributionGoal; LeaderReplicaDistributionGoal; LeaderBytesInDistributionGoal; PreferredLeaderElectionGoal
+
+To reduce complexity, we recommend that you use the inherited master optimization goals, unless you need to _completely_ exclude one or more goals from users. The priority order of the master optimization goals can be modified, if desired, in the default optimization goals (see below).
+
+Master optimization goals are controlled, if needed, in the Cruise Control deployment configuration: `Kafka.spec.cruiseControl.config.goals`
+
+* To accept the inherited master optimization goals, specify an empty value for the `goals` configuration option:
++
+[source=yaml]
+----
+goals: {}
+----
+
+* If you need to modify the inherited master optimization goals, specify a list of goals in the `goals` configuration option.
+
+[id="default-goals"]
+[discrete]
+=== Default optimization goals
+
+We recommend that you always set _default optimization goals_ for the Kafka cluster.
+You can include any or all of the master optimization goals in the default optimization goals.
+
+Default optimization goals are set in the Cruise Control deployment configuration: `Kafka.spec.cruiseControl.config.default.goals`
```

Comment:
I think that this notation in this case is confusing. As we are using this notation it means:

```yaml
spec:
  cruiseControl:
     config:
        default:
            goals:
```

which is of course wrong because the right field is `default.goals` (so it's a field as a whole not `goals` under `default`).
For this reason I would say `default.goals` under `Kafka.spec.cruiseControl.config`.

Reply:
Good suggestion, thanks.